xdsl/transforms/riscv_register_allocation.py

```diff
@@ -0,0 +1,47 @@
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.dialects.riscv import (
+    Register,
+    RegisterType,
+    RdRsImmOperation,
+    RdRsRsOperation,
+    RdImmOperation,
+    NullaryOperation,
+)
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    RewritePattern,
+    PatternRewriter,
+    PatternRewriteWalker,
+    op_type_rewrite_pattern,
+)
+
+
+class AllocateRegisters(RewritePattern):
+    idx: int
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.idx = 0
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self,
+        op: NullaryOperation | RdRsImmOperation | RdRsRsOperation | RdImmOperation,
```

Comment:
Would the following work?
```suggestion
        op: RiscvOp,
```

Reply:
I'm not sure that we want to do this in a RewriteWalker. We can't allocate registers in a peephole rewrite in general, so it seems that it might be better off in the `apply` function directly. That way we can easily check that all the operations in the module are RISC-V operations, etc

Reply:
I don't get your comment, is it fixed with the above proposed change?
What should go in the `apply` function directly?